tests/test_label_categories.py

```diff
@@ -0,0 +1,134 @@
+import pytest
+
+from datumaro import LabelCategories
+
+
+def test_add_category():
+    categories = LabelCategories()
+    index = categories.add("cat")
+    assert index == 0
+    assert len(categories) == 1
+    assert categories[0].name == "cat"
+
+
+def test_add_category_with_parent():
+    categories = LabelCategories()
+    index = categories.add("cat", parent="animal")
+    assert index == 0
+    assert len(categories) == 1
+    assert categories[0].name == "cat"
+    assert categories[0].parent == "animal"
+
+
+def test_add_category_with_attributes():
+    categories = LabelCategories()
+    attributes = {"color", "size"}
+    index = categories.add("cat", attributes=attributes)
+    assert index == 0
+    assert len(categories) == 1
+    assert categories[0].attributes == attributes
+
+
+def test_add_duplicate_category():
+    categories = LabelCategories()
+    categories.add("cat")
+    with pytest.raises(KeyError, match="Label '' 'cat' already exists"):
+        categories.add("cat")
+
+    categories.add("cat", parent="animal")
+    with pytest.raises(KeyError, match="Label 'animal' 'cat' already exists"):
+        categories.add("cat", parent="animal")
```

Comment:
Mixing several test cases in a single test call is a bad practice. Please consider using `@pytest.mark.parametrize`.

Reply:
Good point, thanks!